devicetypes/qubino/qubino-flush-shutter.src/qubino-flush-shutter.groovy

```diff
@@ -0,0 +1,501 @@
+	if (state.venetianBlindDni) {
+		def child = childDevices.find { it.deviceNetworkId == state.venetianBlindDni }
+		events.each {
+			child.sendEvent(it)
+		}
+		createEvent(descriptionText: "Venetian Blinds level has been updated")
+	} else {
+		log.warn "There's no venetian child device to send events to"
+	}
+}
+
+private createChildDevice(componentName, componentLabel, dthName, childIt, namespace = "smartthings", isComponent = true) {
+	try {
+		def childDni = "${device.deviceNetworkId}:$childIt"
+		def child = addChildDevice(namespace, dthName, childDni, device.getHub().getId(), [
+				completedSetup: true,
+				label         : componentLabel,
+				isComponent   : isComponent,
+				componentName : componentName,
+				componentLabel: componentLabel
+		])
+		return child
+	} catch(Exception e) {
+		log.debug "Exception: ${e}"
+	}
+}
+
+private createVenetianBlindsChildDeviceIfNeeded() {
+	if (state.currentMode.contains("Venetian")) {
+		state.venetianBlindDni = createChildDevice("venetianBlind", "Venetian Blind", "Child Switch Multilevel", 2).deviceNetworkId
+	}
+}
+
+private getParameterMap() {[
+		[
+				name: "Operating modes", key: "operatingModes", type: "enum",
+				parameterNumber: 71, size: 1, defaultValue: 0,
+				values: [
+						0: "Shutter mode",
+						1: "Venetian mode (up/down and slate rotation)"
+				],
+				description: "By setting this parameter to 2 the device enters the calibration mode. The parameter relevant only if the device is set to work in positioning mode (parameter 151 set to 1, 2 or 4)."
+		],
+		[
+				name: "Slats tilting full turn time", key: "slatsTiltingFullTurnTime", type: "range",
+				parameterNumber: 72, size: 2, defaultValue: 150,
+				range: "0..32767",
+				description: "This parameter defines the time necessary for slats to make full turn (180 degrees). 100 = 1 second "
+		],
+		[
+				name: "Slats position", key: "slatsPosition", type: "enum",
+				parameterNumber: 73, size: 1, defaultValue: 1,
+				values: [
+						0: "Slats return to previously set position only in case of Z-wave control (not valid for limit switch positions)",
+						1: "Slats return to previously set position in case of Z-wave control, push-button operation or when the lower limit switch is reached"
+				],
+				description: "This parameter defines slats position after up/down movement through Z-wave or push-buttons."
+		],
+		[
+				name: "Motor moving up/down time", key: "motorMovingUp/DownTime", type: "range",
+				parameterNumber: 74, size: 2, defaultValue: 0,
+				range: "0..32767",
+				description: "This parameter defines Shutter motor moving time of complete opening or complete closing. Please check manual for full reference."
+		],
+		[
+				name: "Motor operation detection", key: "motorOperationDetection", type: "range",
+				parameterNumber: 76, size: 1, defaultValue: 30,
+				range: "0..127",
+				description: "Power threshold to be interpreted when motor reach the limit switch."
```

Comment:
I'm not sure what this one does

Reply:
When motor reaches one of the limit switches, there's a spike of power usage, I'm not exactly sure how it differs on certain setups, so I thought I'll better leave it as an option. Just in case, if there would be actually a need to adjust that.